python/vowpalwabbit/pyvw.py

```diff
@@ -1354,3 +1356,656 @@ def get_label(self, label_class=simple_label):
+        features=(list, Feature),
+    )
+
+    def __init__(self, features, name=None, value=None):
+        """
+        Initialize a Namespace instance.
+
+        Parameters
+        ----------
+
+        features : Feature or list of Feature
+            A (list of) Feature object(s) that will form the namespace
+        name : str/int/float, optional
+            The name of the namespace
+        value : int/float, optional
+            A constant that specify the scaling factor for the features of this
+            namespace.
+
+        Examples
+        --------
+
+        >>> from pyvw import Namespace, Feature
+        >>> ns_one_feature = Namespace(Feature(Col("a")))
+        >>> ns_multi_features = Namespace([Feature(Col("a")), Feature(Col("b"))])
+        >>> ns_one_feature_with_name = Namespace(Feature(Col("a")),
+                                                 name="FirstNamespace")
+
+        Returns
+        -------
+
+        self: Namespace
+
+        """
+        for (arg, name_arg) in zip([name, value, features], ["name", "value", "features"]) :
+            check_type(arg, self.expected_type[name_arg])
+
+        self.features = (
+            list(features) if isinstance(features, (list, set)) else [features]
+        )
+        if (value is not None) and (name is None):
+            raise ValueError(
+                "Namespace cannot have a 'value' argument without a 'name' argument"
+            )
+        self.name = name
+        if value is not None:
+            value = str(value)
+        self.value = value
+
+    def process(self, df=None):
+        """
+        Returns the Namespace string representation
+        """
+        out = ["|"]
+        if self.name is not None:
+            out += str(self.name)
+            if self.value is not None:
+                out += [":", str(self.value)]
+
+        return "".join(out)
+
+
+class DFtoVW:
+    """
+    The DFtoVW is used to convert a pandas DataFrame to a suitable VW format.
+    Instances of this class are build using Col object(s) and classes such as
+    SimpleLabel, Feature or Namespace.
+    The class also provided a convenience constructor to initialize the class
+    based on the target/features columns names only.
+    """
+
+    def __init__(self, df, namespaces,
+                 label=None, tag=None, base=None, importance=None):
+        """
+        Initialize a DFtoVW instance
+
+        Parameters
+        ----------
+
+        df : pandas.DataFrame
+            The dataframe to convert to VW input format.
+        namespaces : list of Namespace/Namespace
+            One or more Namespace object(s), each of being composed of one or
+            more Feature object(s).
+        label : SimpleLabel
+            The label is the real numbers to be predicted for the examples.
+        importance : SimpleLabel
+            The importance (weight) indicating the relative importance of the
+            examples.
+        tag :  SimpleLabel
```

Comment:
Tag can only be a string

Reply:
I think there is misunderstanding in what I called `SimpleLabel` -maybe I should rename the class-. `SimpleLabel` can be used to specify constant value or column value (that is different for each cell). It does not specify a position (like the position of a label, or the tag).

In the case of tag, I understood that it's a unique identifier for each example. Then it should be __different__ for each example. This won't be the case if you pass tag as a string constant, no ?